### PR TITLE
feat!: Python 3.12 Upgrade

### PR DESCRIPTION
**Pull request by huniafatima-arbi**
**Merge checklist:**
Check off if complete *or* not applicable:
- [x] Documentation updated (not only docstrings)
- [x] Fixup commits are squashed away
- [x] Unit tests added/updated
- [x] Manual testing instructions provided
- [x] Noted any: Concerns, dependencies, migration issues, deadlines, tickets

**Post-merge:**
- [ ] [Backported](https://openedx.atlassian.net/wiki/spaces/COMM/pages/2065367719/Making+a+pull+request+for+a+named+release) to latest and next named releases

### CODE REVIEW

**Review by iamsobanjaved — APPROVED** [review]
LGTM 👍

**Review by attiyaIshaque — APPROVED** [review]
Looks good to me. 👍 Please rebase your branch with the latest change.

File: requirements/base.txt [hunk 1]
```diff
@@ -210,46 +201,42 @@ requests-oauthlib==2.0.0
...
-sqlparse==0.5.0
+sqlparse==0.5.1
     # via django
-stevedore==5.2.0
+stevedore==5.3.0
     # via
     #   edx-django-utils
     #   edx-opaque-keys
-stripe==10.0.0
+stripe==11.2.0
```
Comment (aht007) [inline requirements/base.txt:227]: @attiyaIshaque there is major change in stripe version. Please consult the changelogs to see if there is any breaking change that needs to catered to in our code. Also please keep in mind to test all stripe related functionality on staging too.

File: requirements/common_constraints.txt [hunk 2]
```diff
@@ -18,16 +18,19 @@ Django<5.0
...
-# https://github.com/open-telemetry/opentelemetry-python/issues/3570
-# Normally this could be added as a constraint in edx-django-utils, where we're
-# adding the opentelemetry dependency. However, when we compile pip-tools.txt,
-# that uses version 7.x, and then there's no undoing that when compiling base.txt.
-# So we need to pin it globally, for now.
-# Ticket for unpinning: https://github.com/openedx/edx-lint/issues/407
-importlib-metadata<7
+# Cause: https://github.com/openedx/event-tracking/pull/290
+# event-tracking 2.4.1 upgrades to pymongo 4.4.0 which is not supported on edx-platform.
+# We will pin event-tracking to do not break existing installations
```
Comment (aht007) [inline requirements/common_constraints.txt:29]:
```suggestion
# We will pin event-tracking so it does not break existing installations
```

Comment (aht007) [reply requirements/common_constraints.txt:29]: I see it is coming from common constraints file in which case it should be fixed there and not here

File: requirements/base.txt (hunk 1 above)
Comment (attiyaIshaque) [reply requirements/base.txt:227]: I noticed the Stripe change, @aht007 . So, I have it in mind that we need to test the Stripe functionality at this stage.

Comment (aht007) [reply requirements/base.txt:227]: We also need to review the Changelogs. Can you consult stripe documentation. It is a major change.

**Comment by attiyaIshaque** [comment]
Please resolve the conflicts.

**Comment by huniafatima-arbi** [comment]
> Please resolve the conflicts.

@attiyaIshaque 
Rebased the PR and resolved conflicts. Please proceed with the merge

**Review by aht007 — APPROVED** [review]
LGTM

**Pull request merged**